x/vesting/keeper/integration_test.go

```diff
@@ -256,28 +256,46 @@ var _ = Describe("Clawback Vesting Accounts", Ordered, func() {
 			Expect(delRes.DelegationResponses[0].Balance.Amount).To(Equal(vested[0].Amount))
 		})
 
-		It("can delegate tokens from account balance - tokens not in vesting schedule", func() {
+		It("account with free balance - delegates the free balance amount. It is tracked as locked vested tokens for the spendable balance calculation", func() {
 			testAccount := testAccounts[0]
 
 			// send some funds to the account to delegate
 			coinsToDelegate := sdk.NewCoins(sdk.NewCoin(stakeDenom, math.NewInt(1e18)))
 			err = testutil.FundAccount(s.ctx, s.app.BankKeeper, testAccount.address, coinsToDelegate)
 			Expect(err).To(BeNil())
 
+			// check that coins to delegate are greater than the locked up vested coins
+			Expect(coinsToDelegate.IsAllGT(vested)).To(BeTrue())
```

Comment:
nit: the check should be before sending
```suggestion
			// send some funds to the account to delegate
			coinsToDelegate := sdk.NewCoins(sdk.NewCoin(stakeDenom, math.NewInt(1e18)))
			// check that coins to delegate are greater than the locked up vested coins
			Expect(coinsToDelegate.IsAllGT(vested)).To(BeTrue())
			
			err = testutil.FundAccount(s.ctx, s.app.BankKeeper, testAccount.address, coinsToDelegate)
			Expect(err).To(BeNil())


```